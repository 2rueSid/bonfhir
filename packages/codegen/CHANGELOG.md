# @bonfhir/codegen

## 2.4.0

### Minor Changes

- [`e5dd1d5`](https://github.com/bonfhir/bonfhir/commit/e5dd1d5411f4ae68ecff706f2f0277ab766e7aac) Thanks [@julienblin](https://github.com/julienblin)! - [Housekeeping] - update typescript module resolution to "Bundler"

### Patch Changes

- [`6b8e816`](https://github.com/bonfhir/bonfhir/commit/6b8e8164afea6c06de22bf8e1313b29057a9ff6e) Thanks [@julienblin](https://github.com/julienblin)! - [Housekeeping] - Update dependencies

## 2.3.0

### Minor Changes

- 11b949d: Fix codegen problem with allowed types extensions for OperationParameters

### Patch Changes

- 9b1d478: Updated dependencies, including Prettier/ESLint that touched all files

## 2.2.0

### Minor Changes

- Fix FHIR type generation

## 2.1.0

### Minor Changes

- Multiple fixes to types + dependency bumps
