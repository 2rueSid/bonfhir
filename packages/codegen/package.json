{
  "name": "@bonfhir/codegen",
  "version": "2.0.0",
  "publishConfig": {
    "registry": "https://npm.pkg.github.com"
  },
  "scripts": {
    "bonfhir-codegen": "ts-node src/index.ts",
    "build": "pnpm clean && rollup --config rollup.config.mjs",
    "check": "prettier --check ./src && eslint ./src && tsc --noEmit",
    "clean": "rimraf dist",
    "dev": "ts-node src/index.ts",
    "format": "prettier --loglevel warn --write ./**/*.ts && eslint --fix ./**/*.ts",
    "test": "jest"
  },
  "type": "module",
  "bin": {
    "bonfhir-codegen": "./dist/cjs/index.cjs"
  },
  "exports": {
    "import": "./dist/esm/index.mjs",
    "require": "./dist/cjs/index.cjs"
  },
  "devDependencies": {
    "@bonfhir/eslint-config": "workspace:*",
    "@bonfhir/eslint-plugin": "workspace:*",
    "@bonfhir/prettier-config": "workspace:*",
    "@bonfhir/typescript-config": "workspace:*",
    "@rollup/plugin-commonjs": "^24.1.0",
    "@rollup/plugin-json": "^6.0.0",
    "@rollup/plugin-node-resolve": "^15.0.2",
    "@rollup/plugin-replace": "^5.0.2",
    "@rollup/plugin-terser": "^0.4.3",
    "@rollup/plugin-typescript": "^11.1.0",
    "@swc/core": "^1.3.55",
    "@types/chunk-text": "^1.0.0",
    "@types/ejs": "^3.1.2",
    "@types/jest": "^29.5.1",
    "@types/listr": "^0.14.4",
    "@types/micromatch": "^4.0.2",
    "@types/node": "^18.16.2",
    "@types/yargs": "^17.0.24",
    "chalk": "^5.2.0",
    "chunk-text": "^2.0.1",
    "ejs": "^3.1.9",
    "esbuild-jest": "^0.5.0",
    "fast-glob": "^3.2.12",
    "jest": "^29.5.0",
    "listr": "^0.14.3",
    "micromatch": "^4.0.5",
    "rimraf": "^5.0.0",
    "rollup": "^3.21.0",
    "rollup-plugin-filesize": "^10.0.0",
    "ts-node": "^10.9.1",
    "typescript": "^5.1.6",
    "yargs": "^17.7.2"
  },
  "prettier": "@bonfhir/prettier-config"
}
