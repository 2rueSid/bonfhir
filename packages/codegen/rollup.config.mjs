import commonjs from "@rollup/plugin-commonjs";
import { nodeResolve } from "@rollup/plugin-node-resolve";
import replace from "@rollup/plugin-replace";
import typescript from "@rollup/plugin-typescript";
import { mkdirSync, readFileSync, writeFileSync } from "node:fs";

export default ["cjs", "esm"].map((format) => ({
  input: "src/index.ts",
  output: [
    {
      file: `dist/${format}/index.${format === "cjs" ? "c" : "m"}js`,
      format,
      sourcemap: true,
      banner: "#!/usr/bin/env node",
    },
  ],
  plugins: [
    replace({
      preventAssignment: true,
      values: {
        "process.env.PACKAGE_VERSION": `"${
          JSON.parse(readFileSync("package.json", "utf8")).version
        }"`,
      },
    }),
    nodeResolve({
      exportConditions: ["node"],
    }),
    commonjs(),
    typescript({
      outDir: `dist/${format}`,
      declaration: false,
      declarationMap: false,
    }),
    {
      buildEnd: () => {
        mkdirSync(`./dist/${format}`, { recursive: true });
        writeFileSync(
          `./dist/${format}/package.json`,
          `{"type": "${format === "cjs" ? "commonjs" : "module"}}"}`
        );
      },
    },
  ],
  onwarn(warning, warn) {
    if (["THIS_IS_UNDEFINED", "CIRCULAR_DEPENDENCY"].includes(warning.code))
      return;
    warn(warning);
  },
}));
