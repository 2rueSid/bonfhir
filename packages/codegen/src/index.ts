import chalk from "chalk";
import yargs from "yargs";
import generate from "./commands/generate";

try {
  yargs()
    .scriptName("bonfhir-codegen")
    .demand(1, chalk.red("Error: Must provide a valid command"))
    .help("h")
    .alias("h", "help")
    .strictCommands()
    .command(generate)
    .demandCommand(1)
    .version(process.env.PACKAGE_VERSION || "local")
    .parse(process.argv.slice(2));
} catch (error) {
  console.error(chalk.red(error));
  console.error();
  console.error(chalk.gray((error as Error).stack));
}
