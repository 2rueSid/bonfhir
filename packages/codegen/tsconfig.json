{
  "extends": "@bonfhir/typescript-config/tsconfig.json",
  "include": ["src/**/*"],
  "compilerOptions": {
    "allowSyntheticDefaultImports": true
  },
  "ts-node": {
    "esm": true,
    "transpileOnly": true,
    "experimentalSpecifierResolution": "node"
  }
}
