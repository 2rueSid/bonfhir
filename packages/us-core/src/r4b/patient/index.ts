export * from "./birth-sex";
export * from "./gender-identity";
export * from "./patient";
export * from "./race";
export * from "./sex";
