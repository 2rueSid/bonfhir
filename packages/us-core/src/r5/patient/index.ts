export * from "./birth-sex";
export * from "./ethnicity";
export * from "./gender-identity";
export * from "./patient";
export * from "./race";
export * from "./sex";
export * from "./tribal-affiliation";
