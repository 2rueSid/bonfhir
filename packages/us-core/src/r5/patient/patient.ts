import { usCorePatientBirthSex } from "./birth-sex";
import { usCoreEthnicity } from "./ethnicity";
import { usCorePatientGenderIdentity } from "./gender-identity";
import { usCoreRace } from "./race";
import { usCorePatientSex } from "./sex";
import { usCoreTribalAffiliation } from "./tribal-affiliation";

export function usCorePatient() {
  return {
    race: usCoreRace(),
    ethnicity: usCoreEthnicity(),
    tribalAffiliation: usCoreTribalAffiliation(),
    birthsex: usCorePatientBirthSex(),
    sex: usCorePatientSex(),
    genderIdentity: usCorePatientGenderIdentity(),
  };
}
