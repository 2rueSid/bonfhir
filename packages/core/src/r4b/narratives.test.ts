import conceptMap from "../../fixtures/conceptmap-example.fhir.json";
import organization from "../../fixtures/organization-example.fhir.json";
import patient from "../../fixtures/patient-example.fhir.json";
import { AnyDomainResource } from "./fhir-types.codegen";
import { narrative } from "./narratives.codegen";

describe("narrative", () => {
  it.each(<Array<AnyDomainResource>>[patient, organization, conceptMap])(
    "generate ($resourceType)",
    (resource) => {
      const result = narrative(resource);
      expect(result).toMatchSnapshot();
    }
  );
});
