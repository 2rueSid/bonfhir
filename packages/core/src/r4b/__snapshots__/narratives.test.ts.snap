// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`narrative narrative Organization 1`] = `
{
  "div": "<div xmlns="http://www.w3.org/1999/xhtml"><ul><li><span>Name: </span>Health Level Seven International</li></ul></div>",
  "status": "generated",
}
`;

exports[`narrative narrative Patient 1`] = `
{
  "div": "<div xmlns="http://www.w3.org/1999/xhtml"><ul><li><span>Active: </span>yes</li><li><span>Address: </span>534 Erewhon St PeasantVille, Rainbow, Vic  3999</li><li><span>Birth Date: </span>12/25/1974</li><li><span>Deceased: </span>no</li><li><span>Gender: </span>male</li><li><span>Identifier: </span>urn:oid:1.2.36.146.595.217.0.1: 12345</li><li><span>Managing Organization: </span>Organization/1</li><li><span>Name: </span>Jim, Peter Chalmers, and Peter Windsor</li><li><span>Telecom: </span>(03) 5555 6473, (03) 3410 5613, and (03) 5555 8834</li></ul></div>",
  "status": "generated",
}
`;
